Framework/Core/include/Framework/ConfigParamSpec.h

```diff
@@ -16,13 +16,44 @@
 namespace o2 {
 namespace framework {
 
-struct ConfigParamSpec {
+/// @class ConfigParamSpec Definition of options for a processor
+/// An option definition consists of a name, a type, a help message, and
+/// an optional default value. The type of the argument has to be specified
+/// in terms of VariantType
+///
+/// Options and arguments can be retrieved from the init context in the
+/// initialization function:
+///   context.options().get<TYPE>("NAME")
+///
+/// All options are also forwarded to the device.
+class ConfigParamSpec {
+ public:
   using ParamType = VariantType;
+  ConfigParamSpec(std::string name, ParamType type,
+                  Variant defaultValue, std::string help)
+    : mName(name)
```

Comment:
I think this (making `ConfigParamSpec` a class and adding trivial getter / setter) should be discussed throughly. At the moment it's kinda semantic the fact that all the `WorkflowSpec` related classes are plain structs, both to reduce configuration boilerplate at minimum and to simplify bindings to other languages. While we can of course re-discuss this, I'd like this is done globally (not on a per class basis) and keeping UX of the configuration in mind (e.g. right now everything which is `*Spec` can be initialised with `{}`). Would you mind holding on for this part of the PR? Empty default values are of course welcome.